code/game/gamemodes/shadowling/shadowling.dm

```diff
@@ -0,0 +1,212 @@
+
+
+/datum/game_mode/proc/finalize_shadowling(var/datum/mind/shadow_mind)
+	var/mob/living/carbon/human/S = shadow_mind.current
+	shadow_mind.current.verbs += /mob/living/human/proc/shadowling_hatch
+	if(shadow_mind.assigned_role == "Clown")
+		S << "<span class='notice'>Your alien nature has allowed you to overcome your clownishness.</span>"
+		S.dna.remove_mutation(CLOWNMUT)
+
+
+
+/*
+	GAME FINISH CHECKS
+*/
+
+
+/datum/game_mode/shadowling/check_finished()
+	var/shadows_alive = 0 //and then shadowling was kill
+	for(var/datum/mind/shadow in shadows) //but what if shadowling was not kill?
+		if(!istype(shadow.current,/mob/living/carbon))
+			continue
+		if(shadow.current.stat==2)
+			continue
+		shadows_alive++
+
+	if (shadows_alive)
+		return ..()
+	else
+		shadowling_dead = 1 //but shadowling was kill :(
+		return 1
+
+
+/datum/game_mode/shadowling/proc/check_shadow_victory()
+	var/success = 0 //Did they win?
+	if(shadow_objectives.Find("enthrall"))
+		success = shadowling_ascended
+	return success
+
+
+/datum/game_mode/shadowling/declare_completion()
+	if(check_shadow_victory())
+		world << "<span class='alienalert'><font size=3><b>The shadowlings have ascended and taken over the station!</FONT></b></span>"
+	else if(shadowling_dead)
+		world << "<span class='danger'><font size=3><b>The shadowlings have been killed by the crew!</b></FONT></span>"
+	else if(!check_shadow_victory())
+		world << "<span class='danger'><font size=3><b>The crew has escaped the station before the shadowlings could ascend!</b></FONT></span>"
+	..()
+	return 1
+
+
+/datum/game_mode/proc/auto_declare_completion_shadowling()
+	var/text = ""
+	if(shadows.len)
+		text += "<br><font size=2><b>The shadowlings were:</b></font>"
+		for(var/datum/mind/shadow in shadows)
+			text += printplayer(shadow)
+		if(thralls.len)
+			text += "<br><font size=2><b>The thralls were:</b></font>"
+			for(var/datum/mind/thrall in thralls)
+				text += printplayer(thrall)
+	else
+		world << "YA BROKE IT M8"
+	text += "<br>"
+	world << text
+
+
+/*
+	MISCELLANEOUS
+*/
+
+
+/datum/species/shadow/ling
+	//Normal shadowpeople but with enhanced effects
+	name = "Shadowling"
+	id = "shadowling"
+	say_mod = "chitters"
+	darksight = 8
+	sexes = 0
+	meat = /obj/item/weapon/reagent_containers/food/snacks/meat/human/mutant/shadow
+
+/datum/species/shadow/ling/spec_life(mob/living/carbon/human/H)
+	var/light_amount = 0
+	H.adjustOxyLoss(-25) //Shadowlings don't need oxygen to survive
+	H.nutrition = NUTRITION_LEVEL_WELL_FED //i aint never get hongry
+	if(isturf(H.loc)) //Copypasta
+		var/turf/T = H.loc
+		var/area/A = T.loc
+		if(A)
+			if(A.lighting_use_dynamic)	light_amount = T.lighting_lumcount
+			else						light_amount =  10
+		if(light_amount > 2) //Rapid death while in the light, countered by...
+			H.take_overall_damage(0,15)
+			H << "<span class='userdanger'><font size=3>The light burns you!</font></span>" //huge spam to say "GET THE FUCK OUT"
```

Comment:
perhaps defer this to every few MC ticks instead of every single spec_life() call.
this is not "huge spam" levels this is NO. Levels.